framework/util/page_guard_manager.cpp

```diff
@@ -341,6 +358,9 @@ bool PageGuardManager::CheckSignalHandler()
 void* PageGuardManager::SignalHandlerWatcher(void* args)
 {
     while (instance_->enable_signal_handler_watcher_ &&
+#if defined(__ANDROID__)
```

Comment:
I'd love later for this#if/#endif inside the condition of a while to be broken out somehow, maybe a convenience function or a lambda defined before the while.  But that shouldn't gate this.

Reply:
What if I made `libsigchain_active_` always a member of `PageGuardManager` on all platforms? That would remove some of the sketchier #ifdefs where I break up the `while` or play games with braces